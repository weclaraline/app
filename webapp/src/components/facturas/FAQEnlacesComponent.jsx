import { Box, Card, CardContent, Divider } from "@material-ui/core";
import React from "react";
import { FAQService, LinksService } from '../../api/';
import EnlacesListComponent from './FAQ/EnlacesListComponent';
import FAQListComponent from './FAQ/FAQListComponent';

const faqService = new FAQService();
const linksService = new LinksService();

const FAQEnlacesComponent = () => {
    const [links, setLinks] = React.useState([]);
    const [faqs, setFaqs] = React.useState([]);

    const fetchFAQs = async () => {
        const faqsServiceResponse = await faqService.make(); 
        setFaqs(faqsServiceResponse);
    }

    const fetchLinks = async () => {
        const linksServiceResponse = await linksService.make(); 
        setLinks(linksServiceResponse);
    }
    React.useEffect(() => {
        fetchFAQs();
        fetchLinks();
    }, []);

    return (
        <Box pr={4} data-testid="panel_faq_enlaces">
            <Card>
                <CardContent>
                    <FAQListComponent faqs={faqs}/>

                    <Divider/>

                    <EnlacesListComponent links={links}/>
                </CardContent>
            </Card>
        </Box>
    );
}

export default FAQEnlacesComponent;