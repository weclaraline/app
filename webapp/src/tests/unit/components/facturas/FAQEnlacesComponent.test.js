import "../../testing_utils";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import FAQEnlacesComponent from "../../../../components/facturas/FAQEnlacesComponent";

it("Renders original content", () => {
    const tree = shallow(<FAQEnlacesComponent/>);
    expect(toJson(tree)).toMatchSnapshot();
});