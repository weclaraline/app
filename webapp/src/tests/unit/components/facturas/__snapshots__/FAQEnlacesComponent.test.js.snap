// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<Styled(MuiBox)
  data-testid="panel_faq_enlaces"
  pr={4}
>
  <WithStyles(ForwardRef(Card))>
    <WithStyles(ForwardRef(CardContent))>
      <FAQListComponent
        faqs={Array []}
      />
      <WithStyles(ForwardRef(Divider)) />
      <EnlacesListComponent
        links={Array []}
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</Styled(MuiBox)>
`;
